lib/rmt/cli/systems.rb

```diff
@@ -5,20 +5,38 @@ class RMT::CLI::Systems < RMT::CLI::Base
   option :csv, type: :boolean, desc: _('Output data in CSV format')
 
   def list
-    systems = (options.all ? System.all : System.limit(options.limit)).order(id: :desc)
-    decorator = RMT::CLI::Decorators::SystemDecorator.new(systems)
+    systems = System.limit(options.limit).order(id: :desc)
+    decorator = RMT::CLI::Decorators::SystemDecorator.new(systems) unless options.all
 
     if systems.empty?
       warn _('There are no systems registered to this RMT instance.')
     elsif options.csv
-      puts decorator.to_csv
+      if options.all
+        decorator = RMT::CLI::Decorators::SystemDecorator.new(systems.first, all: true)
+        puts decorator.csv_headers
```

Comment:
Here, instead of making a new instance, you could create a new constant on the decorator class and use it here, like:

```suggestion
puts RMT::CLI::SystemDecorator::CSV_HEADERS
```